xrpl/wallet/main.py

```diff
@@ -1,44 +1,43 @@
 """The information needed to control an XRPL account."""
-
 from __future__ import annotations
 
-from typing import Type
+from typing import Optional, Type
 
-from xrpl import CryptoAlgorithm
 from xrpl.core.keypairs import derive_classic_address, derive_keypair, generate_seed
 
 
 class Wallet:
     """The information needed to control an XRPL account."""
 
-    def __init__(self: Wallet, seed: str) -> None:
+    def __init__(self: Wallet, pub_key: str, priv_key: str) -> None:
         """
         Generate a new Wallet.
 
         Args:
-            seed: The seed from which the public and private keys are derived.
+            pub_key: The public key for this Wallet.
+            priv_key: The private key for this Wallet.
+
         """
-        self.seed = seed
-        self.pub_key, self.priv_key = derive_keypair(self.seed)
+        self.pub_key = pub_key
+        self.priv_key = priv_key
         self.classic_address = derive_classic_address(self.pub_key)
         self.next_sequence_num = 0
 
     @classmethod
-    def create(
-        cls: Type[Wallet], crypto_algorithm: CryptoAlgorithm = CryptoAlgorithm.SECP256K1
-    ) -> Wallet:
+    def create(cls: Type[Wallet], seed: Optional[str] = None) -> Wallet:
         """
         Generates a new seed and Wallet.
 
         Args:
-            crypto_algorithm: The key-generation algorithm to use when generating the
-                seed. Defaults to SECP256K1.
+            seed: A seed from which to derive public and private keys. If none
+                is provided, one will be generated for you automatically (using
+                Ed25519).
 
         Returns:
             The wallet that is generated from the given seed.
         """
-        seed = generate_seed(algorithm=crypto_algorithm)
-        return cls(seed)
+        pub_key, priv_key = derive_keypair(seed or generate_seed())
```

Comment:
why did you default SECP256K1? I _believe_ ED25519 is preferred.

Reply:
I believe that was a part of the code Amie wrote, and I left it as-is because I didn't know that. That's fine.